includes/class-newspack-popups-inserter.php

```diff
@@ -477,6 +477,24 @@ public static function assess_categories_filter( $popup ) {
 		return true;
 	}
 
+	/**
+	 * If Pop-up has tags, it should only be shown on posts/pages with those.
+	 *
+	 * @param object $popup The popup to assess.
+	 * @return bool Should popup be shown based on tags it has.
+	 */
+	public static function assess_tags_filter( $popup ) {
+		$post_tags  = get_the_tags();
+		$popup_tags = get_the_tags( $popup['id'] );
+		if ( $popup_tags && count( $popup_tags ) ) {
+			return array_intersect(
+				array_column( $post_tags, 'term_id' ),
```

Comment:
Maybe this will avoid the error?

```suggestion
				array_column( ! empty( $post_tags ) ? $post_tags : [], 'term_id' ),
```

Reply:
Good catch! Fixed in 5e6d0490a5ab3d81fd09bcc29b5636d3309fe88f
Added a check for categories, too. I was unable to make a post category-less, but this is just for consistency.